core/frontend/src/store/mavlink.ts

```diff
@@ -0,0 +1,64 @@
+import Vue from 'vue'
+import {
+  Action, Module, Mutation, VuexModule,
+} from 'vuex-module-decorators'
+
+import mavlink2rest from '@/libs/MAVLink2Rest'
+import Listener from '@/libs/MAVLink2Rest/Listener'
+import store from '@/store'
+import { Dictionary } from '@/types/common'
+import { MavlinkMessage } from '@/types/mavlink'
+
+interface messsageRefreshRate {
+  message: string
+  refreshRate: number
+}
+
+@Module({
+  dynamic: true,
+  store,
+  name: 'mavlink_store',
+})
+
+export default class MavlinkStore extends VuexModule {
+  available_messages: Dictionary<MavlinkMessage> = {}
+
+  message_listeners: Dictionary<Listener> = {}
+
+  @Action({ commit: 'updateMessage' })
+  setMessageRefreshRate(rate: messsageRefreshRate): void {
+    const { messageName, refreshRate } = rate
+    if (refreshRate < 0) {
+      console.warn(`Invalid request rate requested for message ${messageName}@${refreshRate}Hz`)
+    }
+
+    mavlink2rest.requestMessageRate(messageName, refreshRate)
```

Comment:
Is this async? If so it would be good to await for it.

Reply:
This is not async, it'll only request a message rate to the backend.

Reply:
It's definitely not mandatory, but it would be good for `requestMessageRate` to be async, since there's a post post request there, right?

Reply:
I'm not sure if I follow, what is the post post request ?

Reply:
https://github.com/bluerobotics/companion-docker/blob/6db2eee1dfea7b9ecdc712752bc52f7aa404e3f3/core/frontend/src/libs/MAVLink2Rest/index.ts#L123

Reply:
This is handled inside the function, there is no need to deal with it for every single call of `requestMessageRate`

Reply:
But it's blocking hehe

Reply:
Apparently not